packages/squiggle-lang/src/public/SqValuePath.ts

```diff
@@ -165,6 +165,28 @@ function astOffsetToPathEdges(ast: ASTNode, offset: number): SqValuePathEdge[] {
   return buildRemainingPathEdges(ast);
 }
 
+export class SqValuePathList {
+  constructor(public paths: SqValuePath[]) {}
+
+  difference(other: SqValuePathList | undefined): SqValuePathList {
+    if (other) {
+      return new SqValuePathList(
+        this.paths.filter(
+          (path) => !other.paths.some((otherPath) => path.isEqual(otherPath))
+        )
+      );
```

Comment:
This is arguably a bit inefficient, because the paths should be in order, but that would make things more complex, and the paths shouldn't get too long. 

I guess it can be N^2, can change if you think it will be an issue.